packages/amplify-category-api/src/provider-utils/awscloudformation/cdk-stack-builder/apigw-stack-builder.ts

```diff
@@ -37,7 +40,11 @@ export class AmplifyApigwResourceStack extends cdk.Stack implements AmplifyApigw
    * @param logicalId
    */
   addCfnOutput(props: cdk.CfnOutputProps, logicalId: string): void {
+    if (this._seenLogicalIds.has(logicalId)) {
+      throw new Error(`logical id "${logicalId}" already exists`);
```

Comment:
Since this logic is repeated in a few places, can you move it (the `if` check, as well as adding to `_seenLogicalIds`) to a helper function.